specifyweb/frontend/js_src/lib/components/FormSliders/RecordSelectorFromCollection.tsx

```diff
@@ -0,0 +1,128 @@
+import { relationshipIsToMany } from '../WbPlanView/mappingHelpers';
+import { RecordSelectorProps, RecordSelectorState } from './RecordSelector';
+import { useRecordSelector } from './RecordSelector';
+
+export function RecordSelectorFromCollection<SCHEMA extends AnySchema>({
+  collection,
+  relationship,
+  onAdd: handleAdd,
+  onDelete: handleDelete,
+  onSlide: handleSlide,
+  children,
+  defaultIndex = 0,
+  isInteraction,
+  ...rest
+}: Omit<
+  RecordSelectorProps<SCHEMA>,
+  | 'index'
+  | 'isDependent'
+  | 'onAdd'
+  | 'onDelete'
+  | 'records'
+  | 'relatedResource'
+  | 'table'
+> &
+  Partial<Pick<RecordSelectorProps<SCHEMA>, 'onAdd' | 'onDelete'>> & {
+    readonly collection: Collection<SCHEMA>;
+    readonly relationship: Relationship;
+    readonly defaultIndex?: number;
+    readonly children: (state: RecordSelectorState<SCHEMA>) => JSX.Element;
+    readonly isInteraction?: boolean;
+  }): JSX.Element | null {
+  const getRecords = React.useCallback(
+    (): RA<SpecifyResource<SCHEMA> | undefined> =>
+      Array.from(collection.models),
+    [collection]
+  );
+  const [records, setRecords] =
+    React.useState<RA<SpecifyResource<SCHEMA> | undefined>>(getRecords);
+
+  const isDependent = collection instanceof DependentCollection;
+  const isLazy = collection instanceof LazyCollection;
+  const isToOne =
+    !relationshipIsToMany(relationship) || relationship.type === 'zero-to-one';
+
+  // Listen for changes to collection
+  React.useEffect(
+    () =>
+      resourceOn(
+        collection,
+        'add remove destroy',
+        (): void => setRecords(getRecords),
+        true
+      ),
+    [collection, getRecords]
+  );
+
+  const [index, setIndex] = useTriggerState(Math.max(0, defaultIndex));
+
+  // Fetch records if needed
+  React.useEffect(() => {
+    /*
+     * BUG: make this more efficient (if going to the last record,
+     *   don't need to fetch all records in between)
+     */
+    if (
+      isLazy &&
+      collection.related?.isNew() !== true &&
+      !collection.isComplete() &&
+      collection.models[index] === undefined
+    )
+      collection
+        .fetch()
+        .then(() => setRecords(getRecords))
+        .catch(raise);
+  }, [collection, isLazy, getRecords, index, records.length]);
+
+  const state = useRecordSelector({
+    ...rest,
+    index,
+    table: collection.table.specifyTable,
+    records,
+    relatedResource: isDependent ? collection.related : undefined,
+    totalCount: collection._totalCount ?? records.length,
+    onAdd: (rawResources): void => {
+      const resources = isToOne ? rawResources.slice(0, 1) : rawResources;
+      if (isDependent && isToOne)
+        collection.related?.placeInSameHierarchy(resources[0]);
+      if (!isInteraction) collection.add(resources);
+      handleAdd?.(resources);
+      setIndex(
+        isInteraction ? collection.models.length : collection.models.length - 1
+      );
+      handleSlide?.(collection.models.length - 1, false);
```

Comment:
Careful! handleSlide is being called with "-1" here if isIntearction

I would change this code to this:
```ts
      const lastIndex = Math.max(0, collection.models.length - 1);
      setIndex(lastIndex);
      handleSlide?.(lastIndex, false);
```

That fixes all issue EXCEPT for the fact that the subview will show record one index before the one that will be added by interaction dialog - you can fix it by adding a prop to the prep select dialog